src/identity.rs

```diff
@@ -0,0 +1,117 @@
+            if let Ok(entry) = entry {
+                if let Ok(file_type) = entry.file_type() {
+                    if file_type.is_dir() {
+                        path.push(entry.file_name());
+                        path.push("crt.pem");
+                        let certs = Certificates::load(&path)
+                            .expect("failed to load certificates from crt.pem");
+                        let local_name = entry.file_name().into_string().unwrap();
+                        tracing::info!(?local_name, "added");
+                        identities.insert(local_name, certs);
+                    }
+                }
+            }
+        }
+
+        Ok(IdentityService { identities })
+    }
+
+    pub fn empty() -> IdentityService {
+        let identities = HashMap::new();
+        IdentityService { identities }
+    }
+
+    pub async fn serve(self, addr: impl Into<SocketAddr>) -> Result<(), Error> {
+        let addr = addr.into();
+        let span = tracing::info_span!("IdentityService::serve", listen.addr = %addr);
+        tracing::info!(parent: &span, "Starting identity server...");
+
+        tonic::transport::Server::builder()
+            .trace_fn(|headers| tracing::debug_span!("request", ?headers))
+            .add_service(IdentityServer::new(self))
+            .serve(addr)
+            .instrument(span)
+            .await?;
+
+        Ok(())
+    }
+}
+
+impl Certificates {
+    // Taken from: https://github.com/linkerd/linkerd2-proxy/blob/23995e7fb6eae5ede81048bdf9e4f68f7e81c7a9/linkerd/app/integration/src/identity.rs#L44-L64
+    fn load<P>(path: P) -> Result<Certificates, io::Error>
+    where
+        P: AsRef<Path>,
+    {
+        let file = File::open(path)?;
+        let mut reader = BufReader::new(file);
+        let certs = rustls::internal::pemfile::certs(&mut reader)
+            .map_err(|_| io::Error::new(ErrorKind::Other, "rustls error reading certs"))?;
+        let leaf = certs
+            .get(0)
+            .ok_or_else(|| io::Error::new(ErrorKind::Other, "no certs in pemfile"))?
+            .as_ref()
+            .into();
+        let intermediates = certs[1..].iter().map(|i| i.as_ref().into()).collect();
+        Ok(Certificates {
+            leaf,
+            intermediates,
+        })
+    }
+}
+
+#[tonic::async_trait]
+impl Identity for IdentityService {
+    async fn certify(
+        &self,
+        request: tonic::Request<pb::CertifyRequest>,
+    ) -> Result<tonic::Response<pb::CertifyResponse>, tonic::Status> {
+        let pb::CertifyRequest { identity, .. } = request.into_inner();
+        if let Some(certs) = self.identities.get(&identity) {
+            // Ideally we'd load the `not_after` value from the `crt.pem`, but
+            // that does not seem to be an option with rustls. Therefore,
+            // create a fake expiration.
+            let not_after = SystemTime::now() + Duration::from_secs(666);
```

Comment:
Are there any issues that could potentially result from this? 

666 seconds is only about 11 minutes. If memory serves, I chose it arbitrarily when writing the integration test code, because none of the integration tests will run for longer than 11 minutes. However, for benchmarking and load testing, it's entirely possible that a benchmark run will take longer than 11 minutes. During a long-running benchmark, the proxy might need to re-certify its identity.

 We probably want these certs to live for at least an hour, as a minimum. If we want to continue being silly, we could hard-code it to something like 
```suggestion
            let not_after = SystemTime::now() + Duration::from_secs(666_420);
```
:)